src/main.ts

```diff
@@ -1,48 +1,61 @@
 import Chart from "chart.js/auto";
-import { Rating } from "./rating";
+import { MetricLabel } from "./metrics";
 import "./style.css";
 import { Target } from "./target";
-import { bonusMult, Weapon } from "./weapon";
-import { NORMALIZED_RATINGS } from "./weapon";
+import { Weapon } from "./weapon";
+import { hasBonus, generateMetrics, normalize, WeaponStats } from "./stats";
+import POLEHAMMER from "./weapons/polehammer.json";
+import MESSER from "./weapons/messer.json";
+import MAUL from "./weapons/maul.json";
+import MAUL from "./weapons/maul.json";
+import { AnyObject } from "chart.js/types/basic";
+
+function weaponFromJson(obj: AnyObject): Weapon {
+  return Object.setPrototypeOf(obj, Weapon.prototype);
+}
+
+let ALL_WEAPONS: Weapon[] = [
+  weaponFromJson(POLEHAMMER),
+  weaponFromJson(MESSER),
+  weaponFromJson(MAUL)
+]
```

Comment:
I've only done 3 so far, but I can write a thing to convert all the typescript weapons to the "source of truth" json if this is a path you want to take